core/src/main/java/org/elasticsearch/ResourceAlreadyExistsException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+
+/**
+ * Generic ResourceAlreadyExistsException corresponding to the {@link RestStatus#BAD_REQUEST} status code.
+ * Sub-classes that require a different status code can achieve that overriding {@link #status()}.
+ */
+public class ResourceAlreadyExistsException extends ElasticsearchException {
+
+    public ResourceAlreadyExistsException(String msg, Object... args) {
+        super(msg, args);
+    }
+
+    public ResourceAlreadyExistsException(StreamInput in) throws IOException {
+        super(in);
+    }
+
+    @Override
+    public RestStatus status() {
```

Comment:
That is how I had it originally. Then I had a brain split about whether `IndexShardAlreadyExistsException` is actually a `ResourceAlreadyExistsException`. I am happy to change this, but I'll wait a bit in case more opinions are voiced.